### DISCUSSION TITLE
Timers....

### DISCUSSION THREAD

**Discussion by rofo69** [BigBobbas/ESP32-S3-Box3-Custom-ESPHome#62] [General]
So, after re-designing my idle page (docs were super useful), I want to be able to set a timer via the device.
If I simply say 'set a timer for x minutes', HASS replies with a message, saying that devices are not supported on this device.

So I thought about writing an automation to do it. This in itself is not difficult, but having hass respond to the device when the timer has ended, is.

If I simply use a wait for trigger (timer.idle) in my automation to send message when the timer has finished, it all works, but because the automation doesn't then end until the wait for trigger condition is met, it basically holds the box3 on the listening screen for as long as the timer is running.

The other way is to have another automation listen for the timer.idle trigger, but how would it then know which media player to send the notification back to, using the 'set conversation response' action (which takes no parameters).

Looking on the community forums, there are a ton of posts, with some huge code examples for what should be a relatively simple thing.

TLDR:- How do a tell the box3 that an action has happened if its not immediate (like turning on a light).

**Comment by BigBobbas** [comment]
Hi 
Timers are supported and the config is written. It is in the dev branch and the config is s3b_dev.yaml 
I'm on my phone at the moment and don't have good access. There is a post in github in the discussions section. I think it may even have a coloured banner at the top for timers. 
Hope that helps

Rob

**Reply by rofo69** [reply]
Thanks Rob, I'll take a look.

**Comment by rofo69** [comment]
Yup.. my bad... the timer thing was at the top in big technicoloured writing!

Timers work great. A couple of observations.

If I simply say 'set a timer for x minutes', all works fine, but I can actually do that multiple times, and although HASS only shows the remaining time for the first timer you set, it actually keeps track of all the timers you've set.

For example, I could set a timer (in this order), for 2 minutes, 1 minute and then 30 seconds. Each will then trigger.

Also... how do I set a timer for the second timer... I've tried using the phrase "set timer 2 for x minutes' but it doesnt recognise it.

**Comment by BigBobbas** [comment]
at the moment the tracking only supports tracking the first timer, however you can set multiple timers, I still have this to add to support viewing the other timers. it is still a bit of a work in progress :) But the code is all in the config if you can figure it out ;)

**Comment by rofo69** [comment]
Been playing around with the s3 this afternoon, amazing what you have managed to get it to do.
Just an FYI, I added switch.turn_off: timer_ringing to the red button, as you can press it whilst a timer is ringing, which then takes you to the idle page, and I didn't find anyway to get back to the timer finished screen to then stop the ringing.

Maybe the cleaner approach would be for the timer menu item to take the user to the timer finished page if the timer_ringing is on.

After I typed that reply, I decided I preferred the menu option and added this to the on_click for the timer page:-

      then:
        - if:
            condition:
             switch.is_off: timer_ringing
            then:
              - display.page.show: time_remaining_page
              - component.update: s3_box_lcd
            else:
              - display.page.show: timer_finished_page
              - component.update: s3_box_lcd

**Comment by BigBobbas** [comment]
yes that is a good point , i hadn't come across that issue - so thanks for pointing it out 👍 

I will add that condition in 

Cheers 😃
